### PR TITLE
New Jabberwock editor

### PR DESCRIPTION
**Pull request by dmo-odoo**
This PR is still a WIP. In particular commits have not been cleaned yet.

### CODE REVIEW

File: addons/web/static/src/js/core/smooth_scroll_on_drag.js [hunk 1]
```diff
@@ -356,6 +340,20 @@ const SmoothScrollOnDrag = Class.extend(mixins.ParentedMixin, {
      * @param {Function} onDragStartCallBack The jQuery drag callback.
      */
     _onSmoothDragStart(ev, ui, onDragStartCallBack) {
+        const elementOffset = $(ev.target).offset();
+        this.options.jQueryDraggableOptions.cursorAt = {
```
Comment (IT-Ideas) [inline addons/web/static/src/js/core/smooth_scroll_on_drag.js:396]: Hello!
Previously the cursorAt was set that way only if none where passed in through the options. Why having changed that behavior?

File: addons/web/static/src/js/core/smooth_scroll_on_drag.js [hunk 2]
```diff
@@ -252,13 +250,13 @@ const SmoothScrollOnDrag = Class.extend(mixins.ParentedMixin, {
 
         // If this.$scrollTarget is the html tag, we need to take the scroll position in to account
         // as offsets positions are calculated relative to the document (thus <html>).
+        const scrollTargetScrollTop = this.$scrollTarget.scrollTop();
```
Comment (IT-Ideas) [inline addons/web/static/src/js/core/smooth_scroll_on_drag.js:257]: Hello!
I don't understand the change you made. For me either you move all the code after the condition or you don't move any. And do not forget to change the comment above accordingly 😉 

The history behind those lines is that I had a different behavior in the web_editor and in sign. Those lines made it work everywhere although I did never really understood the reason behind.
Please be sure that you do not break anything.

cc: @tivisse @qsm-odoo

File: addons/web/static/src/js/core/smooth_scroll_on_drag.js [hunk 3]
```diff
@@ -55,7 +55,9 @@ const SmoothScrollOnDrag = Class.extend(mixins.ParentedMixin, {
      * @constructor
      * @param {Object} parent The parent widget that uses this class.
      * @param {jQuery} $element The element the smooth scroll on drag has to be set on.
-     * @param {jQuery} $scrollTarget The element the scroll will be triggered on.
+     * @param {jQuery} $target Container for drop element. $scrollTarget is this closest
+     *        scrollable parent element.
+     * @param {jQuery} $container The element the scroll will be triggered on.
```
Comment (IT-Ideas) [inline addons/web/static/src/js/core/smooth_scroll_on_drag.js]: May I kindly benefit of your commit and ask you to change the type of the params options.scrollBoundaries.*?
I made a mistake in their types 😱 (it is have written Object although they are boolean)...
Thanks 😄

File: addons/web/static/src/js/core/smooth_scroll_on_drag.js (hunk 2 above)
Comment (Gorash) [reply addons/web/static/src/js/core/smooth_scroll_on_drag.js:257]: The system before did not work on all dom structures. We have elements that drag and drop into a box where one (and sometimes more than one) parent has a scroll bar. And the source of the drag & drop being outside one of the scrolls (but in the second when there are several scrolls). It was similarly that did not work if the destination was in an iframe.

File: addons/web/static/src/js/core/smooth_scroll_on_drag.js (hunk 3 above)
Comment (Gorash) [reply addons/web/static/src/js/core/smooth_scroll_on_drag.js]: Indeed, an error, this comment should not be present.

File: addons/web/static/src/js/core/smooth_scroll_on_drag.js (hunk 1 above)
Comment (Gorash) [reply addons/web/static/src/js/core/smooth_scroll_on_drag.js:396]: For a drag and drop in an iframe it didn't work.

Comment (Gorash) [reply addons/web/static/src/js/core/smooth_scroll_on_drag.js:396]: this branch is no longer used, it is the branch: `14.0-jabberwock-sge-age-dmo-chm-nby`

**Comment by dmo-odoo** [comment]
I'm closing this PR in favor of https://github.com/odoo/odoo/pull/59128

**Comment by dmo-odoo** [comment]
@IT-Ideas @Gorash fixed the scrolling of sign (which was indeed broken) on the latest build. Could you please test https://github.com/odoo/odoo/pull/59128 and confirm it still works as you expect it to ?

**Pull request closed without merging**